lib/shopify_api/rest/request.ex

```diff
@@ -53,9 +53,12 @@ defmodule ShopifyAPI.REST.Request do
 
   ## HTTPoison Overrides
 
-  def request(%HTTPoison.Request{method: method, url: url} = request) do
-    {time, response} = :timer.tc(&super/1, [request])
-    log_request(method, url, time, response)
+  def logged_request(method, url, body, headers, options) do
+    {time, response} = :timer.tc(&request/5, [method, url, body, headers, options])
+    token = Keyword.get(options, :token, %ShopifyAPI.AuthToken{})
+
+    log_request(token, method, url, time, response)
```

Comment:
If we're adding the `:telemetry` integration that allows consumers of this library to implement their own request logging, this may be a good time to mark the internal logging as deprecated? Or, since the library is still not at a `1.0`, we could just zap it at the same time we make this change.

Reply:
Without a trivial way of consuming the telemetry, I don't think logging and telemetry fulfill the same purpose. If logging is not turned on at the debug level, there is no additional cost here either.

Reply:
Maybe, but I think I can go either way on it — I think there's more value in the telemetry than the logging for consumers of the library. The logging can be helpful for the development experience  of working with the library, but can also be implemented on top of the telemetry integration relatively trivially to do more sophisticated things.

I think it might at least be valuable to provide a method of disabling the client's internal logging if I, as a user, am implementing something more sophisticated instead with the telemetry data. Does that seem reasonable? Doesn't need to be done in this PR, but it's something I'd like us to consider.

Reply:
I need to read up a bit more on logging in elixir. My presumption was that you could set logging levels fairly granularity (e.g. `debug` for `ShopifyAPI.*`), but I have not done that with Elixir, so that could be a bad assumption. 
I agree that there needs to be a way to turn of logging for the library if there isn't one already.